src/sagas/repository.js

```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
  * Datasets.
  */
 
-export function* createDataset({ payload }) {
+export function* createDataset() {
+  const dataset = yield select(getCreateDataset);
+  const payload = {
+    name: dataset.name,
+    description: dataset.description,
+    contents: [
+      {
+        source: {
+          studyName: dataset.study,
+          assetName: dataset.asset,
+        },
+        rootValues: dataset.ids,
+      },
+    ],
+  };
+  console.log(payload);
   try {
     const response = yield call(authPost, '/api/repository/v1/datasets', payload);
     const jobId = response.data.id;
     yield put({
       type: ActionTypes.CREATE_DATASET_JOB,
-      payload: { data: response, createdDataset: payload },
+      payload: { data: response, createdDataset: { jobId: payload } },
+      // The created dataset will now be defined by the job id
+      // I can't assign though,
+      // I need to add to the createdDataset? so that there is a lookup?
```

Comment:
Yes, I think we should do this in the reducer -- add something to the store for fetching datasetRequests by jobId.